src/ui/bind_handlers.js

```diff
@@ -46,7 +46,10 @@ module.exports = function bindHandlers(map: Map, options: {}) {
     }
 
     function onMouseDown(e: MouseEvent) {
-        map.stop();
+        if (!map._isUserDoubleClick) {
```

Comment:
instead of adding this property to the map object, you could create a `DoubleClickZoomHandler#isActive` method as we do in with some of the other handles (ex. https://github.com/mapbox/mapbox-gl-js/blob/master/src/ui/handler/box_zoom.js#L52), so you could call `map.doubleClickZoom.isActive()` for this conditional

Reply:
Good idea, I'll switch to using `isActive`. As for the test, there aren't any existing tests for the DoubleClickZoomHandler, and I'm kind of a little too busy to figure out how to write one to catch this specific bug. Better someone who knows the test framework well should do it.